src/library-authoring/units/LibraryUnitBlocks.tsx

```diff
@@ -233,7 +234,9 @@ export const LibraryUnitBlocks = ({ preview }: LibraryUnitBlocksProps) => {
   const [hidePreviewFor, setHidePreviewFor] = useState<string | null>(null);
   const { showToast } = useContext(ToastContext);
 
-  const { unitId, readOnly } = useLibraryContext();
+  const { readOnly, showOnlyPublished } = useLibraryContext();
+  const { sidebarComponentInfo } = useSidebarContext();
+  const unitId = sidebarComponentInfo?.id;
```

Comment:
@ChrisChV This is breaking unit page when you select any component inside it and reload. The reason being that `sidebarComponentInfo?.id` is set to component when selected. Getting `unitId` from `useLibraryContext` fixes it, but I might be missing something. Is a reason for doing this?

Reply:
When I wanted to add a unit to a course, in the component picker, when I opened the unit info sidebar, it generated an error in the context. This code fixed that, but when I reverted that code, it seems that it no longer generates the error, so I think it's safe to revert the code :+1: